packages/@react-spectrum/label/stories/HelpText.stories.tsx

```diff
@@ -9,156 +9,128 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-
+import {ComponentMeta, ComponentStoryObj} from '@storybook/react';
 import {Flex} from '@react-spectrum/layout';
+import {Radio, RadioGroup} from '@react-spectrum/radio';
 import React, {useState} from 'react';
 import {SpectrumTextFieldProps} from '@react-types/textfield';
-import {storiesOf} from '@storybook/react';
 import {TextField} from '@react-spectrum/textfield';
 
-storiesOf('HelpText', module)
-  .addParameters({providerSwitcher: {status: 'positive'}})
-  .add(
-    'description',
-    () => render({description: 'Password must be at least 8 characters.'})
-  )
-  .add(
-    'error message',
-    () => render({errorMessage: 'Create a password with at least 8 characters.', validationState: 'invalid'})
-  )
-  .add(
-    'description and error message',
-    () => {
-      let [value, setValue] = useState('');
+type HelpTextStory = ComponentStoryObj<typeof TextField>;
+
+const argTypes = {
+  label: {
+    control: 'text'
+  },
+  description: {
+    control: 'text'
+  },
+  errorMessage: {
+    control: 'text',
+    defaultValue: 'Create a password with at least 8 characters.'
+  },
+  validationState: {
+    control: 'radio',
+    defaultValue: 'valid',
+    options: ['invalid', 'valid', undefined]
+  },
+  isDisabled: {
+    control: 'boolean',
+    defaultValue: false
+  },
+  labelAlign: {
+    control: 'radio',
+    defaultValue: 'start',
+    options: ['end', 'start']
+  },
+  labelPosition: {
+    control: 'radio',
+    defaultValue: 'top',
+    options: ['side', 'top']
+  },
+  width: {
+    control: 'radio',
+    defaultValue: 'top',
+    options: ['100px', '440px', 'var(--spectrum-global-dimension-top, var(--spectrum-alias-top))']
+  }
+};
+
+export default {
+  title: 'HelpText',
+  component: TextField,
+  args: {
+    label: 'Password',
+    description: 'Password must be at least 8 characters.'
+  },
+  argTypes: argTypes
+} as ComponentMeta<typeof TextField>;
+
+export let Default: HelpTextStory = {};
+
+export let WithState: HelpTextStory = {
+  args: {
+    label: 'Empty field',
+    description: 'This input is only valid when it\'s empty.',
+    errorMessage: 'Remove input.'
+  },
+  argTypes: {validationState: {control: {disable: true}}},
+  render: (props) => <TextFieldWithValidationState {...props} />
```

Comment:
same comment
the reason for this is so that stories are actually about our component and types across stories are the same
this should make it easy to track controls as well

Reply:
I tried using <Story /> instead of <TextField /> and I couldn't add the state this story uses. 👎